app/src/main/java/com/example/indoor_volleyball/Adapters/GymAdapter.java

```diff
@@ -99,12 +100,12 @@ public void bind(Gym gym) throws ParseException {
             rbGymRating.setRating(gym.getRating().floatValue());
 
             //TODO add functionality to the clicks for gyn.
-//            tvGymName.setOnClickListener(new View.OnClickListener() {
-//                @Override
-//                public void onClick(View v) {
-//                    ((MainActivity) context).goToProfileTab(post.getUser());
-//                }
-//            });
+            tvGymName.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    ((MainActivity) context).goToGymDetails(gym);
```

Comment:
`((MainActivity) context)` is sketchy.  Instead of putting that method in mainActivity, just use the technique I talked about above:

```
Intent i = GymDetailActivity.newIntent(context, gym)
startActivity(i);
```

Reply:
Casting like this is often an indication that there's a better way.  Casting can cause issues with code re-use (say you wanted to put this adapter in another activity down the road.  If you did, this line would crash)